lib/ui/common/Alert.dart

```diff
@@ -0,0 +1,223 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      type: MaterialType.transparency,
+      child: Center(
+        child: Container(
+          margin: _Constants.alertEdgePadding,
+          decoration: BoxDecoration(
+            borderRadius: _Constants.cornerRadius,
+            color: _style.backgroundColor,
+          ),
+          child: Padding(
+            padding: _Constants.alertInnerPadding,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                Text(
+                  _viewModel.titleText,
+                  style: _style.titleTextStyle,
+                ),
+                SizedBox(
+                  height: _Constants.titleLineSpace,
+                ),
+                Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text(
+                      _viewModel.detailText,
+                      style: _style.detailTextStyle,
+                    ),
+                    SizedBox(
+                      height: _Constants.detailLineSpace,
+                    ),
+                    Row(
+                      children: _buildAction(context),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _buildAction(BuildContext context) {
+    List<Widget> listBuilder = [
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+            title: _viewModel.cancelActionText,
+            onTap: () => Navigator.pop(context),
+          ),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+            height: _Constants.actionHeight,
+            width: _Constants.actionWidth,
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      ),
+    ];
+    listBuilder.add(
+      SizedBox(
+        width: _Constants.actionLineSpace,
+      ),
+    );
+    listBuilder.add(
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+              title: _viewModel.confirmActionText,
+              onTap: () => confirmAndDismiss(context)),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+            height: _Constants.actionHeight,
+            width: _Constants.actionWidth,
+            backgroundColor: _viewModel.isDestructive
+                ? _style.destructiveActionBackgroundColor
+                : _style.actionBackgroundColor,
+            buttonTextStyle: _style.actionTextStyle,
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      ),
+    );
+    return listBuilder;
+  }
+
+  confirmAndDismiss(BuildContext context) {
+    _viewModel.confirmAction();
+    Navigator.pop(context);
```

Comment:
same as above comment. do we need to get the navigator we are in, and not the global one?
maybe we always should, as it will return the global if that it´s its navigator ? @DavidFD005

Reply:
https://github.com/farmsmart/farmsmart-flutter/pull/58#discussion_r307666278

Reply:
this been looked at?

Reply:
Tested and working as intended